### ISSUE TITLE
v0.33.2 throwing dial tcp 172.20.0.1:443: i/o timeout (Fargate / EKS 1.29)

### ISSUE THREAD

**Issue by lostick** [aws/karpenter-provider-aws#5600]
### Description

**Observed Behavior**:

When running `v0.33.2` on Fargate the pods are failing with the following error:
```
panic: failed to setup manager: failed to determine if *v1.Lease is namespaced: failed to get restmapping: failed to get server groups: Get "https://172.20.0.1:443/api": dial tcp 172.20.0.1:443: i/o timeout

goroutine 1 [running]:
github.com/samber/lo.must({0x25cae80, 0xc0005b0b00}, {0xc00099f760, 0x1, 0x1})
	github.com/samber/lo@v1.39.0/errors.go:51 +0x1c5
github.com/samber/lo.Must[...](...)
	github.com/samber/lo@v1.39.0/errors.go:65
sigs.k8s.io/karpenter/pkg/operator.NewOperator()
	sigs.k8s.io/karpenter@v0.33.0/pkg/operator/operator.go:160 +0x13b8
main.main()
	github.com/aws/karpenter/cmd/controller/main.go:33 +0x25
```

Note that we have recently upgraded our EKS clusters to Kubernetes `1.29`.
I wonder if this is related to karpenter [not officially supporting 1.29](https://github.com/aws/karpenter-provider-aws/pull/5576). The previous versions of Karpenter seem to handle `1.29` fine.

**Expected Behavior**:

**Reproduction Steps** (Please include YAML):

**Versions**:
- Chart Version: `v0.33.2`
- Kubernetes Version (`kubectl version`): `1.29`
- AMI: `AL2`

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by engedaam** [comment]
The Karpenter team has not yet released a `1.29` supported version of Karpenter. We hope to release a version of Karpenter that will be supported on k8s `1.29` soon.

**Comment by lostick** [comment]
Thanks @engedaam, I'll wait for the release then.
